lib/test.gi

```diff
@@ -980,36 +980,46 @@ InstallGlobalFunction( "TestPackage", function(pkgname)
 local testfile, str;
 if not IsBound( GAPInfo.PackagesInfo.(pkgname) ) then
     Print("#I  No package with the name ", pkgname, " is available\n");
-    return;
+    return fail;
 elif LoadPackage( pkgname ) = fail then
     Print( "#I ", pkgname, " package can not be loaded\n" );
-    return;
+    return fail;
 elif not IsBound( GAPInfo.PackagesInfo.(pkgname)[1].TestFile ) then
     Print("#I No standard tests specified in ", pkgname, " package, version ",
           GAPInfo.PackagesInfo.(pkgname)[1].Version,  "\n");
-    return;
+    return fail;
 else
     testfile := Filename( DirectoriesPackageLibrary( pkgname, "" ), 
                           GAPInfo.PackagesInfo.(pkgname)[1].TestFile );
     str:= StringFile( testfile );
     if not IsString( str ) then
         Print( "#I Test file `", testfile, "' for package `", pkgname, 
         " version ", GAPInfo.PackagesInfo.(pkgname)[1].Version, " is not readable\n" );
-        return;
+        return fail;
     fi;
     if EndsWith(testfile,".tst") then
         if Test( testfile, rec(compareFunction := "uptowhitespace") ) then
             Print( "#I  No errors detected while testing package ", pkgname,
                    " version ", GAPInfo.PackagesInfo.(pkgname)[1].Version, 
                    "\n#I  using the test file `", testfile, "'\n");
+            return true;
         else
             Print( "#I  Errors detected while testing package ", pkgname, 
                    " version ", GAPInfo.PackagesInfo.(pkgname)[1].Version, 
                    "\n#I  using the test file `", testfile, "'\n");
+            return false;
         fi;
     elif not READ( testfile ) then
         Print( "#I Test file `", testfile, "' for package `", pkgname,
         " version ", GAPInfo.PackagesInfo.(pkgname)[1].Version, " is not readable\n" );
+        return fail;
+    else
+        # At this point, the READ succeeded, but we have no idea what the
+        # outcome of that test was. Hopefully, that file printed a message of
+        # its own and then terminated GAP with a suitable error code (e.g. by
+        # using TestDirectory with exitGAP:=true); in that case we never get
+        # here an all is fine.
+        return "UNKNOWN";
```

Comment:
I'd probably just go `fail` here again, as we don't know the result, and people trying to test all packages might get confused by the sudden string. Also "here an all is fine" -> "here and all is fine".

Reply:
Agreed, and done / fixed